### ISSUE TITLE
[Feature] Add indicator that will show experiment assignment is finished, regardless of trackingCallback is called

### ISSUE THREAD

**Issue by tusmenko** [growthbook/growthbook#1646]
### Description of Feature

Currently, to check if growthbook is fully initialized, we are supposed to read the "`ready`" flag: 
https://docs.growthbook.io/lib/react#waiting-for-features-to-load

I'm using it to ensure that all the experiment data is ready before sending any events to my CDP, which should include experiment and version IDs for further A\B comparison.  

Unfortunately, the `ready` flag shows only Feature initialization readiness but tells you nothing about the Experiment assignment status.

This leads to the events sent without experiment details in them, in case I'm sending events ASAP, like "Page view event".
This corrupts data because I can't later compare the "Page view event" with the following events, like clicks and so on.

The current workaround is to postpone event sending by 1-2 seconds, allowing `trackingCallback` to be fired.

The more proper solution might be saving some flag manually in `trackingCallback`, and checking it later, but unfortunately, `trackingCallback` calls only when an experiment assignment actually happens, not when the experiment assignment is finished with whatever the result is. 

So nothing allows you to track that "Fine, everything is settled up, feels free to fire event".
Which leads to the following sequence:

> isGrowthBookConfigured =>  true
> growthbook.ready => true
> Sending OPENED_PAGE ExperimentId: undefined
> growthbook trackingCallback  => EXPERIMENT_DATA

Proposed solutions:

1) Add an `experimentsReady` flag to the GrowthBook root, indicating that the experiment targeting is finished regardless if the assignment to experiment actually happened

or

2) Include experiment assignment status in the `ready` value meaning.

**Comment by alyssamichelle** [comment]
**Our audit of stale tickets suggests this may already be resolved, or no longer applicable, on the current version — but we couldn't prove it from the code alone.** @tusmenko could you confirm whether you still see this on the latest release? @ahdriel assigning to you to make the final call once they reply (or after a couple of weeks of silence).

**Why:** No `experimentsReady` flag was added (rg in packages/sdk-js, sdk-react, docs). SDK 1.0 (PR #2450, May 2024) changed initialization so `await gb.init()` resolves only after setPayload() has run, which synchronously evaluates and applies all auto (visual/redirect) experiments via _updateAllAutoExperiments and fires trackingCallback; feature-flag experiments are evaluated lazily at the moment a feature is read, so there is no separate 'assignment finished' moment. Related helpers: getCompletedChangeIds(), getDeferredTrackingCalls(), triggerAutoExperiments() in packages/sdk-js/src/GrowthBook.ts.

Suggested next step: Explain that with SDK >=1.0 awaiting init() (or using <FeaturesReady>) guarantees auto-experiments have been evaluated, and feature experiments are assigned on read; close if that satisfies the use case.

If it turns out to be fixed, please close with a note. If it stays open, please update the labels to reflect current state (keep `bug`/`enhancement` accurate, add `roadmap` if it's planned, drop stale ones like `2024-tracked`).

<sub>Part of a sweep of 223 long-quiet issues audited against `main` (Sept 2026). Ping @alyssamichelle if the evidence looks wrong.</sub>
